Description: Do not overwrite variable by subsequent AC_SUBST call but concatenate them
Author: Tobias Frost <tobi@debian.org>
Forwarded: https://sourceforge.net/p/avarice/patches/35/
Last-Update: 2015-06-16
---
This patch header follows DEP-3: http://dep.debian.net/deps/dep3/
diff --git a/configure.ac b/configure.ac
index 0365b39..6d123aa 100644
--- a/configure.ac
+++ b/configure.ac
@@ -150,13 +150,15 @@ AC_ARG_ENABLE(
 		[--disable-warnings],
 		[Disable -Wall -Wextra options]),
 	[case "${enableval}" in
-	      yes) ENABLE_WARNINGS="-Wall -Wextra" ;;
+	      yes) ENABLE_WARNINGS=" -Wall -Wextra" ;;
 	      no)  ENABLE_WARNINGS="" ;;
 	      *) AC_MSG_ERROR(bad value ${enableval} for disable-warnings option) ;;
 	      esac],
-	[ENABLE_WARNINGS="-Wall -Wextra"])
+	[ENABLE_WARNINGS=" -Wall -Wextra"])
 
-AC_SUBST([AM_CPPFLAGS], [$ENABLE_WARNINGS])
+_CXXFLAGS=$ENABLE_WARNINGS
+
+#AC_SUBST([AM_CPPFLAGS], [$AM_CPPFLAGS] [$ENABLE_WARNINGS])
 
 # --enable-target-programming / --disable-target-programming
 AC_ARG_ENABLE(
@@ -165,13 +167,15 @@ AC_ARG_ENABLE(
 		[--enable-target-programming],
 		[Enable programming (downloading) the target from ELF file]),
 	[case "${enableval}" in
-	      yes) ENABLE_TARGET_PROGRAMMING="-DENABLE_TARGET_PROGRAMMING=1" ;;
-	      no)  ENABLE_TARGET_PROGRAMMING="-DENABLE_TARGET_PROGRAMMING=0" ;;
+	      yes) ENABLE_TARGET_PROGRAMMING=" -DENABLE_TARGET_PROGRAMMING=1" ;;
+	      no)  ENABLE_TARGET_PROGRAMMING=" -DENABLE_TARGET_PROGRAMMING=0" ;;
 	      *) AC_MSG_ERROR(bad value ${enableval} for ensable-target-programming option) ;;
 	      esac],
-	[ENABLE_TARGET_PROGRAMMING="-DENABLE_TARGET_PROGRAMMING=0"])
+	[ENABLE_TARGET_PROGRAMMING=" -DENABLE_TARGET_PROGRAMMING=0"])
+
+_CXXFLAGS+=$ENABLE_TARGET_PROGRAMMING
 
-AC_SUBST([AM_CPPFLAGS], [$ENABLE_TARGET_PROGRAMMING])
+AC_SUBST([AM_CPPFLAGS], [$_CXXFLAGS])
 
 if test "x$enable_target_programming" = "xyes"; then
    if test "x$ac_found_bfd" = "xno"; then
